middleware/admin/certificate_v1.py

```diff
@@ -125,18 +125,24 @@ class HSMCertificate:
     VERSION = 1  # Only supported version
     ROOT_ELEMENT = "root"
     ELEMENT_BASE_CLASS = HSMCertificateElement
+    ELEMENT_FACTORY = HSMCertificateElement
 
-    @staticmethod
-    def from_jsonfile(path):
+    @classmethod
+    def from_jsonfile(kls, path):
         try:
             with open(path, "r") as file:
                 certificate_map = json.loads(file.read())
 
             if type(certificate_map) != dict:
                 raise ValueError(
-                    "JSON file must contain an object as a top level element")
+                    "Certificate file must contain an object as a top level element")
 
-            return HSMCertificate(certificate_map)
+            if certificate_map.get("version") not in kls.VERSION_MAPPING:
+                raise ValueError("Invalid or unsupported HSM certificate "
+                                 "version (supported versions are "
```

Comment:
Maybe print the version we received (just to facilitate debugging)?

Reply:
Done.